homeassistant/components/prometheus/__init__.py

```diff
@@ -379,6 +380,20 @@ def _handle_climate(self, state):
                     float(action == current_action)
                 )
 
+        current_mode = state.state
```

Comment:
A bit weird that we are usint `state.state` here instead of a named attribute. Does this actually work?

Reply:
Yes, I totally agree with you here. I tried reading the ATTR_HVAC_MODE attribute, but somehow I failed - it is not implemented properly in all integrations, for example, I use climate integrations from Fritzox, and it was not there. So I decided to fallback to state property, since it is implemented in the parent class and defaults to hvac_mode:

```
    @property
    def state(self) -> str:
        """Return the current state."""
        return self.hvac_mode
```

https://github.com/home-assistant/core/blob/dev/homeassistant/components/climate/__init__.py#L208

If we decided to stick to ATTR_HVAC_MODE attribute, we will have to fix some integrations, at least the Fritzbox one.